test/test_rosbag_storage/src/create_and_iterate_bag.cpp

```diff
@@ -10,71 +10,104 @@
 #include "boost/foreach.hpp"
 #include <gtest/gtest.h>
 
+void create_test_bag(const std::string &filename)
+{
+  rosbag::Bag bag;
+  bag.open(filename, rosbag::bagmode::Write);
 
-TEST(rosbag_storage, create_and_iterate_bag)
+  std_msgs::String str;
+  str.data = std::string("foo");
+
+  std_msgs::Int32 i;
+  i.data = 42;
+
+  bag.write("chatter", ros::Time::now(), str);
+  bag.write("numbers", ros::Time::now(), i);
+
+  bag.close();
+}
+
+const char* bag_filename = "/tmp/rosbag_storage_create_and_iterate_bag.bag";
+
+TEST(rosbag_storage, iterator_copy_constructor)
 {
-  const char* bag_filename = "/tmp/rosbag_storage_create_and_iterate_bag.bag";
-  {
-    rosbag::Bag bag;
-    bag.open(bag_filename, rosbag::bagmode::Write);
-
-    std_msgs::String str;
-    str.data = std::string("foo");
-  
-    std_msgs::Int32 i;
-    i.data = 42;
-  
-    bag.write("chatter", ros::Time::now(), str);
-    bag.write("numbers", ros::Time::now(), i);
-
-    bag.close();
-  }
+  // copy ctor
+  rosbag::Bag bag;
+  bag.open(bag_filename, rosbag::bagmode::Read);
+  rosbag::View view(bag, rosbag::TopicQuery("numbers"));
+  rosbag::View::const_iterator it0 = view.begin();
+  EXPECT_EQ(42, it0->instantiate<std_msgs::Int32>()->data);
+  rosbag::View::const_iterator it1(it0);
+  EXPECT_EQ(it0, it1);
+  EXPECT_EQ(42, it1->instantiate<std_msgs::Int32>()->data);
+  ++it1;
+  EXPECT_NE(it0, it1);
+  EXPECT_EQ(42, it0->instantiate<std_msgs::Int32>()->data);
+}
 
-  {
-    rosbag::Bag bag;
-    bag.open(bag_filename, rosbag::bagmode::Read);
+TEST(rosbag_storage, iterator_copy_assignment)
+{
+  // copy assignment
+  rosbag::Bag bag;
+  bag.open(bag_filename, rosbag::bagmode::Read);
+  rosbag::View view(bag, rosbag::TopicQuery("numbers"));
+  rosbag::View::const_iterator it0 = view.begin();
+  EXPECT_EQ(42, it0->instantiate<std_msgs::Int32>()->data);
+  rosbag::View::const_iterator it1;
+  it1 = it0;
+  EXPECT_EQ(it0, it1);
+  EXPECT_EQ(42, it1->instantiate<std_msgs::Int32>()->data);
+  ++it1;
+  EXPECT_NE(it0, it1);
+  EXPECT_EQ(42, it0->instantiate<std_msgs::Int32>()->data);
+}
+
+TEST(rosbag_storage, create_and_iterate_bag)
```

Comment:
Since the creation has been moved out can you please rename the test to just `iterate_bag`.